lib/node_modules/@stdlib/math/base/complex/flipsign/test/test.js

```diff
@@ -0,0 +1,263 @@
+	ere = data.expected_re;
+	eim = data.expected_im;
+
+	for ( i = 0; i < re.length; i++ ) {
+		ans = cflipsign( re[ i ], im[ i ], y[ i ] );
+		if ( ans[ 0 ] === ere[ i ] && ans[ 1 ] === eim[ i ] ) {
+			t.equal( ans[ 0 ], ere[ i ], 're: '+re[ i ]+'. Expected: '+ere[ i ] );
+			t.equal( ans[ 1 ], eim[ i ], 'im: '+im[ i ]+'. Expected: '+eim[ i ] );
+		} else {
+			delta = abs( ans[ 0 ] - ere[ i ] );
+			tol = EPS * abs( ere[ i ] );
+			t.ok( delta <= tol, 'within tolerance. re: '+re[ i ]+'. im: '+im[ i ]+'. Actual re: '+ans[ 0 ]+'. Expected re: '+ere[ i ]+'. delta: '+delta+'. tol: '+tol+'.' );
+
+			delta = abs( ans[ 1 ] - eim[ i ] );
+			tol = EPS * abs( eim[ i ] );
+			t.ok( delta <= tol, 'within tolerance. re: '+re[ i ]+'. im: '+im[ i ]+'. Actual im: '+ans[ 1 ]+'. Expected im: '+eim[ i ]+'. delta: '+delta+'. tol: '+tol+'.' );
+		}
+	}
+	t.end();
+});
+
+tape( 'the function evaluates the flipsign function (output array)', function test( t ) {
+	var expected;
+	var actual;
+	var out;
+
+	// Tested against Julia...
+	out = new Array( 2 );
+	actual = cflipsign( out, -4.2, 5.5, -8 );
+
+	expected = [ 4.2, -5.5 ];
+
+	t.deepEqual( actual, expected, 'returns expected value' );
+	t.strictEqual( actual, out, 'returns output value' );
+
+	t.end();
+});
+
+tape( 'the function evaluates the flipsign function (output typed array)', function test( t ) {
+	var expected;
+	var actual;
+	var out;
+
+	// Tested against Julia...
+	out = new Float64Array( 2 );
+	actual = cflipsign( out, 9.99999, 0.1, 9 );
+
+	expected = new Float64Array( [ 9.99999, 0.1 ] );
+
+	t.deepEqual( actual, expected, 'returns expected value' );
+	t.strictEqual( actual, out, 'returns output value' );
+
+	t.end();
+});
+
+tape( 'the function evaluates the flipsign function (output object)', function test( t ) {
+	var expected;
+	var actual;
+	var out;
+
+	// Tested against Julia...
+	out = {};
+	actual = cflipsign( out, 4.2, -5.5, -7 );
+
+	expected = {
+		'0': -4.2,
+		'1': 5.5
+	};
+
+	t.deepEqual( actual, expected, 'returns expected value' );
+	t.strictEqual( actual, out, 'returns output value' );
+
+	t.end();
+});
+
+tape( 'the function does not flip sign if provided NaN `y`', function test( t ) {
+	var expected;
+	var actual;
+	var out;
+
+	// Tested against Julia...
+	out = new Array( 2 );
+	actual = cflipsign( out, -4.2, 5.5, NaN );
+
+	expected = [ -4.2, 5.5 ];
+
+	t.deepEqual( actual, expected, 'returns expected value' );
+	t.strictEqual( actual, out, 'returns output value' );
+
+	t.end();
+});
+
+tape( 'the function does not flip sign if provided `+infinity` as `y`', function test( t ) {
+	var expected;
+	var actual;
+	var out;
+
+	// Tested against Julia...
+	out = new Array( 2 );
+	actual = cflipsign( out, -4.2, 5.5, PINF );
+
+	expected = [ -4.2, 5.5 ];
+
+	t.deepEqual( actual, expected, 'returns expected value' );
+	t.strictEqual( actual, out, 'returns output value' );
+
+	t.end();
+});
+
+tape( 'the function flips sign if provided `-infinity` as `y`', function test( t ) {
```

Comment:
"the function flips the sign of real and imaginary input arguments if `y` is `-infinity`"